### DISCUSSION TITLE
Error - Cannot read Properties of null (reading '1')

### DISCUSSION THREAD

**Discussion by amiga-500** [aclap-dev/video-downloadhelper#164] [Help and Q&A]
### Your question

So when i attempt to select my "E:\\" drive as the download dir i get the error:

**_Cannot read Properties of null (reading '1')_**

1. My other drives like: C, D, R, Z work just fine.
2. I am using Windows 11.  This problem only started after updating this extension on Firefox two updates ago.  Prior to this, this was not a problem.
3. E:\\ drive is my default download dir for Firefox.  If i change the path to e.g. E:\Test it will work fine.
4. Interestingly, when if i set the download dir to "E:\\" via the settings file and import it, "Default Download Dir" now correctly shows as "E:\\", but when i download a video i get this error:

```moz-extension://b55dfc6a-468e-4261-b546-1af7664ba78c/background/main.js:1
receive@moz-extension://b55dfc6a-468e-4261-b546-1af7664ba78c/background/main.js:1:13157
Jp</callCatchAppNotFound/</<@moz-extension://b55dfc6a-468e-4261-b546-1af7664ba78c/background/main.js:1:42564
```
5. I should also point out in Windows the E:\ path is my OS's Download dir. 
6. I switched from Companion App to Browser and the same error occurs.

Right now the only solution is to downgrade back two versions.

![www](https://github.com/aclap-dev/video-downloadhelper/assets/5191497/9d184dc8-557d-4fed-bed3-1b773f1f3fdc)

### Addon & CoApp version and details

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-CA
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\MyUserName",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by GrampaWildWilly** [comment]
Take a look at #112.  It seems there is something strange about the system downloads directory.  I've not been able to replicate it on my system, but now you're reporting the same problem as that other user.  I'm not sure this is a VDH issue.  It might be, but I'm not sure, especially since I can't replicate it.  Maybe it's a system issue.  I honestly don't know.  I assume that since you've relocated your directory from C:\Users\<user>\Downloads to a whole other partition, that taking ownership (in the sense of the Microsoft jargon) is not an issue.  I would be curious to know if using the new VDH setting on the Behavior tab for the native filepicker makes any difference.

**Reply by GrampaWildWilly** [reply]
Also, 2.0.10 is not the latest CoApp.  Look in the Wiki (link at the top of this page) for the CoApp downloads page.

**Reply by amiga-500** [reply]
Thanks for the reply.  And thanks for the heads up about upgrading my CoApp.  Yea i too am suspecting it's some unusual system issue.  My security settings for the drive is correct (e.g. it includes my user id with full permissions), however.  If i give it a test directory e.g. E:\\Temp the problem disappears.  I would prefer just E:\\.  Firefox saves just fine to this directory for all downloads.  It is only VDH that has this issue.   **The problem disappears if i downgrade VDH back to v8.1.4.0**  So i'll likely just do that and be done with it.  I will obviously have to disable automatic updating of this extension so the problem never returns though, but seems like the best option.

>  if using the new VDH setting on the Behavior tab for the native filepicker makes any difference

In my screencast above i clicked the "Change" link to activate its filepicker.  I assume a soon-to-be-released newer VDH will change this so it now uses a native (windows) filepicker instead?  Or am missing something??

Anyway, no worries.  I will just downgrade to get this fixed now and call it a day.  All good.  Thanks for your help though.

Regards.

**Reply by GrampaWildWilly** [reply]
The latest Firefox beta uses the Windows file selector if you set the right VDH setting.  I see you are using Firefox on Windows so you're in luck.  Just look for the betas in the Wiki same as you did for the CoApp.
